src/lib/components/collections/tab-bar/TabBar.svelte

```diff
@@ -0,0 +1,127 @@
+<script lang="ts">
+  // ---- SVG
+  import plusIcon from "$lib/assets/actionicon-normal.svg";
+  import angleLeft from "$lib/assets/angleLeft.svg";
+  import angleRight from "$lib/assets/angle-right.svg";
+
+  // ---- Store
+  import { collapsibleState } from "$lib/store/request-response-section";
+
+  // ---- Interface
+  import type { TabDocument } from "$lib/database/app.database";
+  import type { NewTab } from "$lib/utils/interfaces/request.interface";
+
+  // ---- Component
+  import Tab from "../req-res-section/sub-components/sub-components-header/Tab.svelte";
+
+  // ---- Helper
+  import { moveNavigation } from "$lib/utils/helpers/navigation";
+
+  export let tabList: TabDocument[] = [];
+  export let onNewTabRequested: () => void;
+  export let onTabClosed: (id: string, tab: NewTab) => void;
+  export let onDropEvent: (event: Event) => void;
+  export let onDragStart: (index: number) => void;
+  export let onDropOver: (index: number) => void;
+  export let onTabSelected: (id: string) => void;
+
+  $: {
+    if (tabList) {
+      if (tabList.length >= 0 && tabList.length <= 5) {
+        tabWidth = 196;
+      } else if (tabList.length >= 6 && tabList.length <= 10) {
+        tabWidth = 140;
+      } else {
+        tabWidth = 100;
+      }
+    }
+  }
+
+  let tabWidth: number = 196;
+  let scrollerParent: number;
+  let scrollerWidth: number;
+
+  const handleDropOver = (event: Event) => {
+    event.preventDefault();
```

Comment:
we discussed there can be a shorthand for this.

Reply:
Done